flink-table/flink-table-planner/pom.xml

```diff
@@ -356,103 +351,130 @@ under the License.
 			<plugin>
 				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-shade-plugin</artifactId>
+				<configuration>
+					<!-- Base config -->
+					<createDependencyReducedPom>true</createDependencyReducedPom>
+					<dependencyReducedPomLocation>${project.basedir}/target/dependency-reduced-pom.xml</dependencyReducedPomLocation>
+					<filters combine.children="append">
+						<filter>
+							<artifact>*:*</artifact>
+							<excludes>
+								<!-- Excluded all these files for a clean flink-table-planner JAR -->
+								<exclude>org-apache-calcite-jdbc.properties</exclude>
+								<exclude>common.proto</exclude>
+								<exclude>requests.proto</exclude>
+								<exclude>responses.proto</exclude>
+								<exclude>codegen/**</exclude>
+								<exclude>META-INF/*.SF</exclude>
+								<exclude>META-INF/*.DSA</exclude>
+								<exclude>META-INF/*.RSA</exclude>
+								<exclude>META-INF/services/java.sql.Driver</exclude>
+								<exclude>META-INF/versions/11/module-info.class</exclude>
+								<!-- com.ibm.icu:icu4j includes a LICENSE file in its root folder -->
+								<exclude>LICENSE</exclude>
+							</excludes>
+						</filter>
+					</filters>
+					<artifactSet>
+						<includes combine.children="append">
+							<include>org.apache.calcite:*</include>
+							<include>org.apache.calcite.avatica:*</include>
+
+							<!-- Calcite's dependencies -->
+							<include>com.esri.geometry:esri-geometry-api</include>
+							<include>com.google.guava:guava</include>
+							<include>com.google.guava:failureaccess</include>
+							<include>commons-codec:commons-codec</include>
+							<include>commons-io:commons-io</include>
+
+							<!-- flink-table-planner dependencies -->
+							<include>org.apache.flink:flink-sql-parser</include>
+							<include>org.apache.flink:flink-sql-parser-hive</include>
+
+							<!-- For legacy string expressions in Table API -->
+							<include>org.scala-lang.modules:scala-parser-combinators_${scala.binary.version}</include>
+
+							<!-- ReflectionsUtil -->
+							<include>org.reflections:reflections</include>
```

Comment:
Isn't this only used for tests? What is it used for? JSON plan?

Reply:
Check `ReflectionsUtil`. It's used for the JSON plan but perhaps can be removed (not in this PR though)